File: 0001_0100/0045/0045.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.read())
    if n > 1:
        a = [0] * 10
        for i in range(9, 1, -1):
            while n % i == 0:
                n //= i
                a[i] += 1
        s = "".join((str(i) * a[i]) for i in range(2, 10)) if n == 1 else "-1"
    else:
        s = "1" if n else "10"
    q.write(s)

File: 0001_0100/0045/0045.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n int
	var s string
	fmt.Fscan(f, &n)
	if n == 0 {
		s = "10"
	} else if n == 1 {
		s = "1"
	} else {
		var a [10]int
		for i := 9; i > 1; i-- {
			for n%i == 0 {
				n /= i
				a[i]++
			}
		}
		if n > 1 {
			s = "-1"
		} else {
			for i := 2; i < 10; i++ {
				s += strings.Repeat(fmt.Sprint(i), a[i])
			}
		}
	}
	fmt.Fprint(q, s)
}
